src/lib/GRPCClient.ts

```diff
@@ -0,0 +1,68 @@
+import { loadSync, PackageDefinition } from '@grpc/proto-loader'
+import { Client, credentials, loadPackageDefinition } from 'grpc'
+
+export class GRPCClient {
+	private packageDefinition: PackageDefinition
+	private client: Client
+	private listNameMethods: any[]
+
+	constructor(
+		protoPath: string,
+		packageName: string,
+		service: string,
+		host: string,
+		options = {} as any,
+		tls: boolean = false
+	) {
+		this.packageDefinition = loadSync(protoPath, {
+			defaults: options.default === undefined ? true : options.default,
+			enums: options.enums === undefined ? String : options.enums,
+			keepCase: options.keepCase === undefined ? true : options.keepCase,
+			longs: options.longs === undefined ? String : options.longs,
+			oneofs: options.default === undefined ? true : options.default,
+		})
+
+		const proto = loadPackageDefinition(this.packageDefinition)[packageName]
+		const listMethods = this.packageDefinition[`${packageName}.${service}`]
+		const channelCredentials = tls
+			? credentials.createSsl()
+			: credentials.createInsecure()
+		this.client = new proto[service](host, channelCredentials)
+		this.listNameMethods = []
+
+		for (const key in listMethods) {
+			if (key) {
```

Comment:
Is it our intention to skip the first method? Or should this be if (listMethods[key])? If so should comment why and use key === 0

Reply:
you are right, it should be if(listMethods[key])